### PR TITLE
feature: dymint manages the tendermint validator set

### PR DESCRIPTION
**Pull request by mtsitrin**
This PR makes dymint manage the validator set instead of using the App ValidatorUpdates

Dymint ignores the validator updates it will receive on `EndBlock`
It will load the validator set based on the hub on Genesis

### CODE REVIEW

File: block/manager.go [hunk 1]
```diff
@@ -88,10 +89,11 @@ type Manager struct {
...
 func getInitialState(store store.Store, genesis *tmtypes.GenesisDoc) (types.State, error) {
 	s, err := store.LoadState()
-	if err != nil {
-		s, err = types.NewFromGenesisDoc(genesis)
+	if err == nil {
+		return s, nil
 	}
-	return s, err
+
+	return types.NewFromGenesisDoc(genesis)
```
Comment (omritoptix) [inline block/manager.go:96]: Seems like if we get an error not related to not having a genesis state, we'll still try to create a genesis state (Like happend to us before) without knowing the root cause of the error until debugging. 
My suggestion is to check for a specific error type if possible before creating `NewFromGensisDoc`

File: state/executor.go [hunk 2]
```diff
@@ -80,7 +74,7 @@ func (e *BlockExecutor) InitChain(genesis *tmtypes.GenesisDoc) (*abci.ResponseIn
 				AppVersion: params.Version.AppVersion,
 			},
 		},
-		Validators:    tmtypes.TM2PB.ValidatorUpdates(tmtypes.NewValidatorSet(validators)),
+		Validators:    tmtypes.TM2PB.ValidatorUpdates(tmtypes.NewValidatorSet(nil)),
```
Comment (omritoptix) [inline state/executor.go:77]: why are we through the app passing nil validator to the abci app vs passing the active sequencer as the validator?
i.e the validator_set is solely determined by dymint but still being passed to the app. 

How does the app know how to distribute rewards for relevant sequencers?

**Comment by omritoptix** [comment]
The linter falls. make sure you have the pre-commit hook setup and running to avoid such cases.

File: state/executor.go (hunk 2 above)
Comment (mtsitrin) [reply state/executor.go:77]: Currently, the app doesn't handle this
The sequencer on the rollapp is registered by tx / genesis for rewards distribution.

**Comment by omritoptix** [comment]
@mtsitrin is the change backward compatible? did you try upgrading a rollapp with this change?

**Comment by mtsitrin** [comment]
> @mtsitrin is the change backward compatible? did you try upgrading a rollapp with this change?

yes. it will remain with last validatorSet it have

**Review by omritoptix — APPROVED** [review]

**Pull request merged**